atlasdb-client/src/main/java/com/palantir/atlasdb/util/AtlasDbMetrics.java

```diff
@@ -51,6 +51,15 @@ private AtlasDbMetrics() {}
 
     public static synchronized void setMetricRegistries(MetricRegistry metricRegistry,
             TaggedMetricRegistry taggedMetricRegistry) {
+        if (!metricRegistry.equals(metrics.get())) {
```

Comment:
I wonder if this is a legitimate place to use `==` rather than `.equals`?  Right now this is what `.equals` does anyway but this makes it clear that if someone is going to set this multiple times they must do so with exactly the same object.

If the `.equals` method was changed to say that 2 MetricsRegistries are equal if they have the same metrics then you could create 2 which are equal during initialisation because they have no metrics but which then diverge if people call methods on them directly.